docs/explanation/components.md

```diff
@@ -0,0 +1,76 @@
+# Understanding Piraeus Datastore Components
+
+Piraeus Datastore consists of seven different components. Every component is running as a separate Deployment, DaemonSet or plain Pod.
+
+All Pods are labelled according to their component. You can check the Pods running in your own cluster by
+running the following `kubectl` command:
+
+```
+$ kubectl get pods '-ocustom-columns=NAME:.metadata.name,COMPONENT:.metadata.labels.app\.kubernetes\.io/component'
+NAME                                                   COMPONENT
+ha-controller-vd82w                                    ha-controller
+k8s-10.test                                            linstor-satellite
+linstor-controller-6c8f8dc47-cm8hr                     linstor-controller
+linstor-csi-controller-59b9968b86-ftl76                linstor-csi-controller
+linstor-csi-node-hcmk9                                 linstor-csi-node
+piraeus-operator-controller-manager-6dcfcb4568-6jntp   piraeus-operator
+piraeus-operator-gencert-59449cb449-nzg6z              piraeus-operator-gencert
+```
+
+# `piraeus-operator`
+
+The Piraeus Operator creates and maintains the other components, except for the `piraeus-operator-gencert` component.
+
+Along with deploying the needed Kubernetes resources, it maintains the LINSTOR® Cluster state by:
+
+* Registering satellites
+* Creating storage pools
+* Maintaining node labels
+
+# `piraeus-operator-gencert`
+
+The "generate certificate" Pod creates and maintains the TLS key and certificate used by the Piraeus Operator.
+The TLS secret, named `webhook-server-cert`, is needed to start the Piraeus Operator Pod. In addition, this Pod
+keeps the `ValidatingWebhookConfiguration` for the Piraeus Operator up-to-date.
+
+The Piraeus Operator needs a TLS certificate to serve the validation endpoint for the custom resources it maintains.
+
+Historically, TLS certificates where created by `cert-manager`. This was removed to reduce the number of dependencies
+for deploying Piraeus Datastore.
+
+# `linstor-controller`
+
+The LINSTOR Controller is responsible for resource placement, resource configuration, and orchestration of any
+operational processes that require a view of the whole cluster. It maintains a database of all the configuration
+information for the whole cluster, stored as Kubernetes objects.
+
+The LINSTOR Controller connects to the LINSTOR Satellites and sends them instructions for achieving the desired cluster
+state.
+
+It provides an external API used by LINSTOR CSI and Piraeus Operator to change the cluster state.
+
+# `linstor-satellite`
```

Comment:
Perhaps mention why the satellites are created as plain pods? It is unusual.

Reply:
Done